common/util/filter_slice_test.go

```diff
@@ -0,0 +1,113 @@
+/*
+Copyright 2022-2023 Nutanix, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"errors"
+	"reflect"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestFilter(t *testing.T) {
+	type args struct {
+		items []int
+		fn    func(item int) bool
+	}
+	tests := []struct {
+		name string
+		args args
+		want []int
+	}{
+		{
+			name: "Test 1: Filter returns a result slice based on the provided function",
+			args: args{
+				items: []int{1, 2, 3, 4, 5, 6, 7, 8, 9, 10},
+				fn:    func(item int) bool { return item%2 == 0 },
+			},
+			want: []int{2, 4, 6, 8, 10},
+		},
+		{
+			name: "Test 2: Filter returns an empty slice if no elements filter through the provided function",
+			args: args{
+				items: []int{1, 2, 3, 4, 5, 6, 7, 8, 9, 10},
+				fn:    func(item int) bool { return item > 9999 },
+			},
+			want: []int{},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := Filter(tt.args.items, tt.args.fn); !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("Filter() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestFindFirstInt(t *testing.T) {
```

Comment:
I think we should rename the core function to FindFirstIntFromFilter, what do you think?

Reply:
The core function is a generic function which is called `Filter` and can work with any data type.  I just tested it with an integer type here because golang did not allow me to use mixed types in tests.

Reply:
ah makes sense.